tests/unitary/test_withdraw_unitary.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python3
+
+import brownie
+
+
+# Can user A and user B withdraw in correct proportion?
+def test_withdraw_multiples(multi, base_token, reward_token, alice, bob, charlie, chain):
+    reward_token.approve(multi, 10 ** 16, {"from": alice})
+
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+
+    amount = 10 ** 12
+    base_token.approve(multi, amount, {"from": bob})
+    multi.stake(amount, {"from": bob})
+    bob_stake_timestamp = multi.lastTimeRewardApplicable(reward_token)
+
+    base_token.approve(multi, 10 * amount, {"from": charlie})
+
+    # chain.mine(timedelta=1)
+    multi.stake(amount * 10, {'from': charlie})
+    charlie_stake_timestamp = multi.lastTimeRewardApplicable(reward_token)
+
+    if charlie_stake_timestamp == bob_stake_timestamp:
+        assert multi.earned(charlie, reward_token) // 10 == multi.earned(bob, reward_token)
+    else:
+        assert multi.earned(charlie, reward_token) // 10 <= multi.earned(bob, reward_token)
+    assert multi.balanceOf(charlie) // 10 == multi.balanceOf(bob)
+
+    bob_init_base_balance = base_token.balanceOf(bob)
+    bob_init_reward_balance = reward_token.balanceOf(bob)
+    bob_exit = multi.exit({"from": bob})
+    bob_exit_val = bob_exit.events["Withdrawn"].values()[1]
+    bob_final_base_balance = base_token.balanceOf(bob)
+    bob_final_reward_balance = reward_token.balanceOf(bob)
+    bob_final_base_gain = bob_final_base_balance - bob_init_base_balance
+    bob_final_reward_gain = bob_final_reward_balance - bob_init_reward_balance
+
+    charlie_init_base_balance = base_token.balanceOf(charlie)
+    charlie_init_reward_balance = reward_token.balanceOf(charlie)
+    charlie_exit = multi.exit({"from": charlie})
+    charlie_exit_val = charlie_exit.events["Withdrawn"].values()[1]
+    charlie_final_base_balance = base_token.balanceOf(charlie)
+    charlie_final_reward_balance = reward_token.balanceOf(charlie)
+    charlie_final_base_gain = charlie_final_base_balance - charlie_init_base_balance
+    charlie_final_reward_gain = charlie_final_reward_balance - charlie_init_reward_balance
+
+    assert charlie_exit_val // 10 == bob_exit_val
+    if charlie_stake_timestamp == bob_stake_timestamp:
+        assert charlie_final_reward_gain // 10 == bob_final_reward_gain
+    else:
+        assert charlie_final_reward_gain // 10 <= bob_final_reward_gain
```

Comment:
same comment as above..  `<=` is too broad.  better as something like `assert 0.999 < (charlie_balance // 10) / bob_balance <= 1`